File: go/client/service/book_service.go
package service

import (
	"github.com/csandiego/poc-openapitools/go/client/data"
)

type BookService interface {
	Create(data.Book) error
	Delete(int) error
	Get(int) (data.Book, error)
	List() ([]data.Book, error)
	Update(int, data.Book) error
}

File: kotlin/Client/src/main/kotlin/com/github/csandiego/pocopenapitools/client/service/BookService.kt
package com.github.csandiego.pocopenapitools.client.service

import com.github.csandiego.pocopenapitools.client.data.Book

interface BookService {
    fun list(): List<Book>
    fun create(book: Book)
    fun get(id: Int): Book
    fun update(id: Int, book: Book)
    fun delete(id: Int)
}
